src/syft/ast/globals.py

```diff
@@ -7,17 +8,23 @@ class Globals(Module):
 
     """The collection of frameworks held in a global namespace"""
 
-    def __init__(self):
+    def __init__(self) -> None:
         super().__init__("globals", None, None, None)
 
-    def add_framework(self, attr_name, attr):
+    def add_framework(self, attr_name: str, attr: object) -> None:
         self.attrs[attr_name] = attr
 
-    def add_path(self, path, return_type_name=None, framework_reference=None):
+    def add_path(
+        self,
+        path: Union[str, List[str]],
+        index: int = 0,
+        return_type_name: Optional[str] = None,
+        framework_reference: Optional[object] = None,
+    ) -> None:
         if isinstance(path, str):
             path = path.split(".")
 
-        framework_name = path[0]
+        framework_name = path[index]
```

Comment:
I.... don't think this is right

Reply:
The framework_name shoudl always be the first part of the path.

Reply:
Yeah this was tough, so the old code had different interfaces in globals vs module which mypy complained about. So i changed it so they matched (which is fine since we try to enforce kwargs anyway), but set the default index to 0 for globals.

```
# globals.py
def add_path(
        self,
        path: Union[str, List[str]],
        index: int = 0,
        return_type_name: Optional[str] = None,
        framework_reference: Optional[object] = None,
    ) -> None:
        if isinstance(path, str):
            path = path.split(".")

        framework_name = path[index]

# module.py
def add_path(
        self,
        path: List[str],
        index: int,
        return_type_name: Optional[str] = None,
        framework_reference: Optional[object] = None,
    ) -> None:
        if path[index] not in self.attrs:

            attr_ref = getattr(self.ref, path[index])
```

Without this index will be unused but also required to match the parent signature.... so not sure what the best solution here is. I was hesitant to refactor too much without solid tests in place.